Formula/amp.rb

```diff
@@ -19,6 +19,7 @@ class Amp < Formula
   uses_from_macos "libiconv"
 
   def install
+    ENV["CFLAGS"] = "-arch #{Hardware::CPU.arch}" if Hardware::CPU.arm?
```

Comment:
```suggestion
    ENV.append_to_cflags "-arch #{Hardware::CPU.arch}" if Hardware::CPU.arm?
```
I know it's not necessary here, but would be good to keep this uniform with places where it is, I think.

Reply:
```
  CMake Error at /opt/homebrew/Cellar/cmake/3.20.1/share/cmake/Modules/CMakeTestCCompiler.cmake:66 (message):
    The C compiler

      "/opt/homebrew/Library/Homebrew/shims/mac/super/clang"

    is not able to compile a simple test program.
```
I hope this change didn't cause the error. Would be an awkward `brew` bug 😅

Reply:
Hmm, nope, didn't go away.

Reply:
```
==> cargo install --locked --root /opt/homebrew/Cellar/amp/0.6.2 --path .
  Installing amp v0.6.2 (/private/tmp/amp-20210423-355-15noiy6/amp-0.6.2)
    Updating crates.io index
warning: package `miow v0.2.1` in Cargo.lock is yanked in registry `crates.io`, consider running without --locked
warning: package `net2 v0.2.33` in Cargo.lock is yanked in registry `crates.io`, consider running without --locked
   Compiling libc v0.2.62
   Compiling log v0.4.8
   Compiling cfg-if v0.1.3
   Compiling pkg-config v0.3.11
   Compiling bitflags v1.1.0
   Compiling num-traits v0.2.4
   Compiling serde v1.0.66
   Compiling winapi-build v0.1.1
   Compiling memchr v2.3.0
   Compiling unicode-xid v0.1.0
   Compiling num-integer v0.1.38
   Compiling byteorder v1.2.3
   Compiling rustc-demangle v0.1.8
   Compiling same-file v1.0.2
   Compiling lazy_static v1.2.0
   Compiling matches v0.1.6
   Compiling safemem v0.2.0
   Compiling plist v0.2.4
   Compiling winapi v0.2.8
   Compiling dtoa v0.4.2
   Compiling unicode-normalization v0.1.7
   Compiling linked-hash-map v0.5.1
   Compiling lazycell v1.2.1
   Compiling regex-syntax v0.6.14
   Compiling itoa v0.4.1
   Compiling slab v0.4.1
   Compiling fnv v1.0.6
   Compiling percent-encoding v2.1.0
   Compiling regex-syntax v0.4.2
   Compiling arc-swap v0.3.11
   Compiling block v0.1.6
   Compiling regex-syntax v0.3.9
   Compiling unicode-width v0.1.5
   Compiling unicode-segmentation v1.0.3
   Compiling fragment v0.3.1
   Compiling utf8-ranges v0.1.3
   Compiling luthor v0.1.7
   Compiling glob v0.2.11
   Compiling yaml-rust v0.3.5
   Compiling smallvec v0.4.5
   Compiling app_dirs v1.2.1
   Compiling kernel32-sys v0.2.2
   Compiling proc-macro2 v0.4.6
   Compiling unicode-bidi v0.3.4
   Compiling thread_local v1.0.1
   Compiling walkdir v2.1.4
   Compiling base64 v0.8.0
   Compiling yaml-rust v0.4.0
   Compiling pad v0.1.5
   Compiling quote v0.6.3
   Compiling idna v0.2.0
   Compiling bloodhound v0.5.4
   Compiling aho-corasick v0.7.8
   Compiling syn v0.14.2
   Compiling num_cpus v1.10.1
   Compiling jobserver v0.1.17
   Compiling time v0.1.40
   Compiling malloc_buf v0.0.6
   Compiling iovec v0.1.2
   Compiling backtrace v0.3.8
   Compiling net2 v0.2.33
   Compiling memchr v0.1.11
   Compiling signal-hook-registry v1.0.1
   Compiling termion v1.5.1
   Compiling xml-rs v0.7.0
   Compiling url v2.1.0
   Compiling thread-id v2.0.0
   Compiling regex v1.3.4
   Compiling objc v0.2.2
   Compiling cc v1.0.45
   Compiling chrono v0.4.2
   Compiling error-chain v0.10.0
   Compiling error-chain v0.12.0
   Compiling mio v0.6.16
   Compiling aho-corasick v0.5.3
   Compiling bincode v1.0.1
   Compiling serde_json v1.0.19
   Compiling serde_derive v1.0.66
   Compiling thread_local v0.2.7
   Compiling objc_id v0.1.0
   Compiling cmake v0.1.31
   Compiling libz-sys v1.0.25
   Compiling miniz-sys v0.1.10
   Compiling libgit2-sys v0.9.1
   Compiling mio-uds v0.6.7
   Compiling amp v0.6.2 (/private/tmp/amp-20210423-355-15noiy6/amp-0.6.2)
   Compiling regex v0.1.80
   Compiling objc-foundation v0.1.1
   Compiling onig_sys v68.0.1
   Compiling signal-hook v0.1.9
   Compiling clipboard v0.4.6
   Compiling flate2 v1.0.1
   Compiling onig v3.2.2
   Compiling syntect v2.1.0
   Compiling scribe v0.7.2
   Compiling git2 v0.10.1
warning: unnecessary parentheses around type
  --> src/util/token.rs:14:38
   |
14 | ...                   -> Option<(Position)> {
   |                                 ^^^^^^^^^^ help: remove these parentheses
   |
   = note: `#[warn(unused_parens)]` on by default

warning: trait objects without an explicit `dyn` are deprecated
  --> src/presenters/modes/search_select.rs:12:66
   |
12 | pub fn display<T: Display>(workspace: &mut Workspace, mode: &mut SearchSelectMode<T>, view: &mut View) -> Result<()> {
   |                                                                  ^^^^^^^^^^^^^^^^^^^ help: use `dyn`: `dyn SearchSelectMode<T>`
   |
   = note: `#[warn(bare_trait_objects)]` on by default

warning: trait objects without an explicit `dyn` are deprecated
   --> src/view/terminal/termion_terminal.rs:337:42
    |
337 | fn map_style(style: Style) -> Option<Box<Display>> {
    |                                          ^^^^^^^ help: use `dyn`: `dyn Display`

warning: trait objects without an explicit `dyn` are deprecated
  --> src/view/terminal/mod.rs:35:43
   |
35 | pub fn build_terminal() -> Result<Arc<Box<Terminal + Sync + Send + 'static>>> {
   |                                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: use `dyn`: `dyn Terminal + Sync + Send + 'static`

warning: trait objects without an explicit `dyn` are deprecated
  --> src/view/buffer/renderer.rs:35:19
   |
35 |     terminal: &'a Terminal,
   |                   ^^^^^^^^ help: use `dyn`: `dyn Terminal`

warning: trait objects without an explicit `dyn` are deprecated
  --> src/view/buffer/renderer.rs:42:41
   |
42 |     scroll_offset: usize, terminal: &'a Terminal, theme: &'a Theme,
   |                                         ^^^^^^^^ help: use `dyn`: `dyn Terminal`

warning: trait objects without an explicit `dyn` are deprecated
   --> src/view/buffer/renderer.rs:219:86
    |
219 |     pub fn render(&mut self, lines: LineIterator<'p>, mut lexeme_mapper: Option<&mut LexemeMapper>) -> Result<Option<Position>> {
    |                                                                                      ^^^^^^^^^^^^ help: use `dyn`: `dyn LexemeMapper`

warning: trait objects without an explicit `dyn` are deprecated
  --> src/view/buffer/scrollable_region.rs:11:23
   |
11 |     terminal: Arc<Box<Terminal + Sync + Send + 'static>>,
   |                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: use `dyn`: `dyn Terminal + Sync + Send + 'static`

warning: trait objects without an explicit `dyn` are deprecated
  --> src/view/buffer/scrollable_region.rs:16:34
   |
16 |     pub fn new(terminal: Arc<Box<Terminal + Sync + Send + 'static>>) -> ScrollableRegion {
   |                                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: use `dyn`: `dyn Terminal + Sync + Send + 'static`

warning: trait objects without an explicit `dyn` are deprecated
 --> src/view/event_listener.rs:8:23
  |
8 |     terminal: Arc<Box<Terminal + Sync + Send + 'static>>,
  |                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: use `dyn`: `dyn Terminal + Sync + Send + 'static`

warning: trait objects without an explicit `dyn` are deprecated
  --> src/view/event_listener.rs:16:36
   |
16 |     pub fn start(terminal: Arc<Box<Terminal + Sync + Send + 'static>>, events: Sender<Event>, killswitch: Receiver<()>) {
   |                                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: use `dyn`: `dyn Terminal + Sync + Send + 'static`

warning: trait objects without an explicit `dyn` are deprecated
  --> src/view/presenter.rs:74:135
   |
74 | ...n<&[Range]>, lexeme_mapper: Option<&'p mut LexemeMapper>) -> Result<()> {
   |                                               ^^^^^^^^^^^^ help: use `dyn`: `dyn LexemeMapper`

warning: trait objects without an explicit `dyn` are deprecated
  --> src/view/mod.rs:38:23
   |
38 |     terminal: Arc<Box<Terminal + Sync + Send + 'static>>,
   |                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: use `dyn`: `dyn Terminal + Sync + Send + 'static`

warning: use of deprecated associated function `std::error::Error::description`: use the Display impl or to_string()
 --> src/errors.rs:2:1
  |
2 | / error_chain! {
3 | |     foreign_links {
4 | |         Io(::std::io::Error) #[cfg(unix)];
5 | |     }
6 | | }
  | |_^
  |
  = note: `#[warn(deprecated)]` on by default
  = note: this warning originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)

warning: use of deprecated associated function `std::error::Error::cause`: replaced by Error::source, which can support downcasting
 --> src/errors.rs:2:1
  |
2 | / error_chain! {
3 | |     foreign_links {
4 | |         Io(::std::io::Error) #[cfg(unix)];
5 | |     }
6 | | }
  | |_^
  |
  = note: this warning originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)

warning: 15 warnings emitted

    Finished release [optimized] target(s) in 1m 12s
  Installing /opt/homebrew/Cellar/amp/0.6.2/bin/amp
   Installed package `amp v0.6.2 (/private/tmp/amp-20210423-355-15noiy6/amp-0.6.2)` (executable `amp`)
warning: be sure to add `/opt/homebrew/Cellar/amp/0.6.2/bin` to your PATH to be able to run the installed binaries
/opt/homebrew/Library/Homebrew/shims/scm/git --version
==> Cleaning
==> Fixing /opt/homebrew/Cellar/amp/0.6.2/bin/amp permissions from 755 to 555
==> Finishing up
ln -s ../Cellar/amp/0.6.2/bin/amp amp
/usr/bin/sandbox-exec -f /private/tmp/homebrew20210423-1746-yelcbe.sb nice ruby -W1 -I $LOAD_PATH -- /opt/homebrew/Library/Homebrew/postinstall.rb /opt/homebrew/Library/Taps/homebrew/homebrew-core/Formula/amp.rb
==> Summary
🍺  /opt/homebrew/Cellar/amp/0.6.2: 8 files, 4.2MB, built in 1 minute 18 seconds
```

Reply:
Not really sure what the difference is between my local environment and the CI. I'll try deleting my cargo cache and see if that changes anything.

Reply:
Nope. Even pulling your suggested change locally and forcefully cleaning out the cache so it's all fetched fresh didn't trigger any failure 🙃.

```
    Finished release [optimized] target(s) in 2m 42s
  Installing /opt/homebrew/Cellar/amp/0.6.2/bin/amp
   Installed package `amp v0.6.2 (/private/tmp/amp-20210423-2300-1uc3ncc/amp-0.6.2)` (executable `amp`)
warning: be sure to add `/opt/homebrew/Cellar/amp/0.6.2/bin` to your PATH to be able to run the installed binaries
/opt/homebrew/Library/Homebrew/shims/scm/git --version
==> Cleaning
==> Fixing /opt/homebrew/Cellar/amp/0.6.2/bin/amp permissions from 755 to 555
==> Finishing up
ln -s ../Cellar/amp/0.6.2/bin/amp amp
```

Reply:
CI is haunted 👻

Reply:
I just pushed a commit using Cargo's `-vv` flag so it should give us a better idea of what's going on... hopefully.